---
title: "RoseArrayTools_vignette"
author: "Jeekin Lau"
date: "4/22/2021"
output:   
  html_document:
    toc: true
    toc_float: 
      collapsed: false
---


# Introduction
This package will be a collection of tools that are routinely used by Texas A&M University Rose Breeding Program for analysis of genetic data from the WagRhSNP68k SNP array.


# Installation
If you have not yet installed RoseArrayTools in R, use the following code to do so.

```{r eval=FALSE, include=T}
install.packages('devtools')
devtools::install_github("jeekinlau/RoseArrayTools")
```


# Functions
**Below are the list of functions currently available:**      

   'compare_probes'      
   'dosage_to_ACGT'     
   'call_specs'    
   'polyorigin_imputation'



## 'compare_probes'
This function 'compare_probes' takes as the input file the .dat long form file that fitPoly produces. However, on the WagRhSNP 68K SNP array, each marker is probed twice (forward and reverse strand). fitPoly scores all these probes as independent markers so we need to compare the probes. This function looks at each marker by looking at the genotype calls at each of the two probes for that given marker. The result of this function prints two new files, a compared_calls.csv file and a compared_calls_kind_counts.csv. 

This function takes a very long time. If you set progress=T or by default it is T, it will tell you which step it is on. Steps 1-3 are just background steps to start the comparison.  The steps comparing_probes and Marker_stats take a while longer and when the function is done it will print "FINISHED" in the console. Depending on speed of computer can take a couple of hours to finish. There are a Total of 5 automated steps with Marker_stats being the last step.

```{r eval=FALSE, include=T}
compare_probes(data_dat="path_to_file.dat", progress=T)
```

## 'dosage_to_ACGT'
This function 'dosage_to_ACGT' requires a csv file returned from fitPoly formatted like below (left) and converts it to what you see on the right. The A allele is the "reference" allele and the B allele is the alternative allele and dosage is defined as the number of copies of the alternative allele.

This is how to use the function. Be aware, this function was written as a loop inside of a loop with 5 ifelse statements thus on large files with many individuals it will take a long time. However there the console will print the marker number it is currently on to give you an estimate to the amount of time left. Default for progress is TRUE however if progress is set at F it will not print the markers completed and will only print a message of "FINISHED" when done. 

**IMPORTANT:** if you leave the separator argument blank, then a the output will not have any sort of delimiter between the characters i.e. a dosage of "1" with reference allele "A" and alternative allele "C" will return the value AAAC however if you designate the separator argument with any character, the output will use that to separate the alleles. For example if you used the same example above and you chose a separator = "|" then the result will be A|A|A|C instead of AAAC. The default is no separator.

```{r eval=FALSE, include=T}
dosage_to_ACGT(dosage.file="path_to_dosage_file.csv", progress=T, separator = "")
```


![image of input file and output file formats no separator](C:/Users/Jeekin/Dropbox/Projects/R_Packages/Dosage_to_ACGT/DoseToACGT/Screenshot 2021-04-02 151131.png)

![image of input file and output file formats with separator="|"](C:/Users/Jeekin/Dropbox/Projects/R_Packages/Dosage_to_ACGT/DoseToACGT/Screenshot 2021-04-23 104929.png)

## 'call_specs'
This function uses an output file of the function 'compare_probes' it uses the compared_calls_kind_counts.csv file to count what percentage of the calls are called by two probes, one probe, different probes, or not called. 

```{r eval=FALSE, include=T}
call_specs(kind_counts_file = "compared_calls_kinds.csv", select_genotypes=NULL)
```
Running the above function results in the output below. The script counts the number of  "S" "O" "D" and NA are in the entire matrix. The script will print a percentage and an estimate of number of markers in each group. both these numbers are approximations as it counts the number of "S" "O" "D" and NA are in the entire matrix and divides by total number of rows x columns in the matrix. 

By default, select_genotypes is NULL which counts and displays the specs of all the genotypes in the file provided. If a .csv file is provided with the genotype's names only those will be selected to be counted.
```{r eval=FALSE, include=T}
[1] "percent______same 0.109028854404337 7455"
[1] "percent_______one 0.359272009497065 24564"
[1] "percent_different 0.0783920744846818 5360"
[1] "percent________NA 0.453307061613916 30994"
```

![image of select_genotypes input file ](C:/Users/Jeekin/Dropbox/Projects/R_Packages/Dosage_to_ACGT/DoseToACGT/Screenshot 2021-04-29 111127.png)


## 'polyoroigin_imputation'
This function takes the polyorigin postdoseprob.csv output which contains conditional probabilities and outputs a csv with the imputed dosages.
For this script function you must provide input.file name, number.parents, ploidy, probability.    
input.file - should be the name of the .csv file you want to convert      
num.parents - number of parents in the .csv file
ploidy - ploidy number as an integer
probability - probability you want to use as a cutoff

```{r eval=FALSE, include=T}
polyorigin_imputation(input.file = "example.csv", number.parents = 3, ploidy = 2, probability = 0.9)
```

![input and output files ](C:/Users/Jeekin/Dropbox/Projects/R_Packages/Screenshot 2021-05-21 100729.png)
